lighthouse-core/gather/gatherers/form-elements.js

```diff
@@ -0,0 +1,154 @@
+ *  @param {HTMLFormElement} formElement
+ *  @return {Array<HTMLElement>}
+ */
+/* istanbul ignore next */
+function getChildrenInputs(formElement) {
+  if (formElement.formless == true){
+    return formElement.inputs
+  }
+
+  const inputsArray = [];
+  const childrenArray = Array.prototype.slice.call(formElement.childNodes);
+  
+  childrenArray.map(  /** @param {HTMLElement} node */ (element) => {
+    if (element.nodeName == 'INPUT' || element.nodeName == 'SELECT' || element.nodeName == 'TEXTAREA'){
+      const inputAttributes = {
+        id: element.id,
+        nodeName: element.nodeName,
+        name: element.name,
+        placeholder: element.placeholder,
+        autocomplete: element.autocomplete,
+      }
+      inputsArray.push(inputAttributes);
+    }
+  });
+
+  return inputsArray;
+}
+
+/**
+ *  @param {HTMLFormElement} formElement
+ *  @return {Array<HTMLElement>}
+ */
+/* istanbul ignore next */
+function getChildrenLabels(formElement) {
+  if (formElement.formless == true){
+    return formElement.labels
+  }
+  const childrenArray = Array.prototype.slice.call(formElement.childNodes);
+  const labelsArray = [];
+  childrenArray.map(/** @param {HTMLElement} node */ (element) => {
+    if (element.nodeName == 'LABEL'){
+      const labelAttributes = {
+        id: element.id,
+        nodeName: element.nodeName,
+        name: element.name,
+        for: element.for,
+      }
+      labelsArray.push(labelAttributes);
+    }
+  });
+
+  return labelsArray;
+}
+
+
+/**
+ * @return {LH.Artifacts['FormElements']}
+ */
+/* istanbul ignore next */
+function collectFormElements() {
+
+  const formElements = getElementsInDocument('form'); // eslint-disable-line no-undef
+  const formChildren = getElementsInDocument('textarea', 'input', 'label', 'select'); // eslint-disable-line no-undef
+
+  const formless = {
+    formless: true,
+    inputs: [],
+    labels: [],
+  }
+  formChildren.map(/** @param {HTMLElement} node */ (childElement) => {
+    if (childElement.form == "" || !childElement.form){
+      if (childElement.nodeName == 'INPUT' || childElement.nodeName == 'SELECT' || childElement.nodeName == 'TEXTAREA'){
+        const inputAttributes = {
+          id: childElement.id,
+          nodeName: childElement.nodeName,
+          name: childElement.name,
+          placeholder: childElement.placeholder,
+          autocomplete: childElement.autocomplete,
+        }
+        formless.inputs.push(inputAttributes);
+      }
+
+      else if (childElement.nodeName == 'LABEL'){
+        const labelAttributes = {
+          id: childElement.id,
+          nodeName: childElement.nodeName,
+          name: childElement.name,
+          for: childElement.for,
+        }
+        formless.labels.push(labelAttributes);
+      }
+    }
+  });
+
+  if (formless.inputs.length > 0 || formless.labels.length > 0){
+    formElements.push(formless);
+  }
+
+  return formElements.map(/** @param {HTMLFormElement} formElement */ (formElement) => {
```

Comment:
Can you assign this to a variable and then append the `formless` entry to it?

Reply:
assign inputs and labels lists as variables?

Reply:
instead of pushing the formless entry before the mapping (L118-120), do something like:

```
const forms = formElements.map( ... );
forms.push(formless);
return forms;
```

Reply:
wow that makes so much sense! got it